src/Jackett.Common/Indexers/NCore.cs

```diff
@@ -51,280 +64,272 @@ public NCore(IIndexerConfigurationService configService, WebClient wc, Logger l,
-            pairs.Add(new KeyValuePair<string, string>("tipus", "kivalasztottak_kozott"));
-            pairs.Add(new KeyValuePair<string, string>("submit.x", "1"));
-            pairs.Add(new KeyValuePair<string, string>("submit.y", "1"));
-            pairs.Add(new KeyValuePair<string, string>("submit", "Ok"));
-            pairs.Add(new KeyValuePair<string, string>("mire", searchString));
+            var pairs = new NameValueCollection
+            {
+                {"nyit_sorozat_resz", "true"},
+                {"tipus", "kivalasztottak_kozott"},
+                {"submit.x", "1"},
+                {"submit.y", "1"},
+                {"submit", "Ok"}
+            };
+            if (query.IsImdbQuery)
+            {
+                pairs.Add("miben", "imdb");
+                pairs.Add("mire", query.ImdbID);
+            }
+            else
+            {
+                pairs.Add("miben", "name");
+                pairs.Add("mire", episodeString == null ? query.GetQueryString() : query.SanitizedSearchTerm);
+            }
 
             var cats = MapTorznabCapsToTrackers(query);
-
             if (cats.Count == 0)
                 cats = GetAllTrackerCategories();
-
-            foreach (var lcat in LanguageCats)
-            {
-                if (!configData.Hungarian.Value)
-                    cats.Remove(lcat + "_hun");
-                if (!configData.English.Value)
-                    cats.Remove(lcat);
-            }
-
-            foreach (var cat in cats)
-                pairs.Add(new KeyValuePair<string, string>("kivalasztott_tipus[]", cat));
-
-            var results = await PostDataWithCookiesAndRetry(SearchUrl, pairs);
+            if (!configData.Hungarian.Value)
+                cats.RemoveAll(cat => cat.Contains("_hun"));
+            if (!configData.English.Value)
+                cats = cats.Except(_languageCats).ToList();
 
 
+            pairs.Add("kivalasztott_tipus[]", string.Join(",", cats));
+            var results = await PostDataWithCookiesAndRetry(SearchUrl, pairs.ToEnumerable(true));
             var parser = new HtmlParser();
             var dom = parser.ParseDocument(results.Content);
-            var numVal = 0;
 
             // find number of torrents / page
             var torrentPerPage = dom.QuerySelector(".box_torrent_all")?.QuerySelectorAll(".box_torrent").Length ?? 0;
             if (torrentPerPage == 0)
                 return releases;
             var startPage = (query.Offset / torrentPerPage) + 1;
-            var previouslyParsedOnPage = query.Offset - (startPage * torrentPerPage) + 1; //+1 because indexing start from 0
-            if (previouslyParsedOnPage < 0)
-                previouslyParsedOnPage = query.Offset;
+            var previouslyParsedOnPage = query.Offset % torrentPerPage;
+
+            // find page links in the bottom
+            var lastPageLink = dom.QuerySelector("div[id=pager_bottom]")?.QuerySelectorAll("a")
```

Comment:
You can combine all selectors and select in one CSS selector:
`div[id=pager_bottom] a[href*=odal]` 
It's not tested!